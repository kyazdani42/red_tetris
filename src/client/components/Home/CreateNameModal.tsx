import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../redux/store';
import {
  CreateRoomStyle,
  ErrorStyle,
  InputWrapper,
  LabelStyle,
  ModalStyle,
  ModalWrapper,
  NameInputStyle
} from './styles';

interface Props {
  playerName: string | null;
  handleDispatch: (username: string) => void;
  setDisplayModal: (d: boolean) => void;
}

export const CreateNameModal: React.SFC<Props> = ({ handleDispatch, setDisplayModal, playerName }) => {
  const [error, setError] = React.useState(null);
  if (playerName) {
    handleDispatch(playerName);
    return null;
  }
  return (
    <ModalWrapper onClick={handleRemoveModal(setDisplayModal)} id="modal-name">
      <ModalStyle>
        <InputWrapper>
          <LabelStyle>Pick a username</LabelStyle>
          <NameInputStyle onKeyDown={handleKeyDown(handleDispatch, setError)} autoFocus={true} />
          {error ? <ErrorStyle>{error}</ErrorStyle> : null}
        </InputWrapper>
        <CreateRoomStyle onClick={handleCreateRoom(handleDispatch, setError)}>
          Launch Game
        </CreateRoomStyle>
      </ModalStyle>
    </ModalWrapper>
  );
};

const handleRemoveModal = (setDisplayModal: any) => (e: any) => {
  if (e.target.id === 'modal-name') {
    setDisplayModal(false);
  }
};

const handleCreateRoom = (dispatch: (username: string) => void, setError: any) => (e: any) => {
  const value = e.target.previousSibling.firstChild.nextElementSibling.value;
  if (!value.length) {
    setError('please enter something');
  } else {
    dispatch(value);
  }
};

const handleKeyDown = (dispatch: (username: string) => void, setError: any) => (e: any) => {
  setError(null);
  if (e.key === 'Enter') {
    if (!e.target.value.length) {
      setError('please enter something');
    } else {
      dispatch(e.target.value);
    }
  }
};

const mapStateToProps = (state: State, ownProps: any) => ({
  playerName: state.app.playerName,
  ...ownProps
});

export default connect(mapStateToProps)(CreateNameModal);
