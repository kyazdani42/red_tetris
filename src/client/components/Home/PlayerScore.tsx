import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../redux/store';
import { PlayerScoreDisplay, PlayerScoreWrapper } from './styles';

interface Props {
  playerScore: BestScore | null;
}

export const PlayerScore: React.SFC<Props> = ({ playerScore }) =>
  (playerScore && (
    <PlayerScoreWrapper>
      <PlayerScoreDisplay>Hello {playerScore.name}</PlayerScoreDisplay>
      <PlayerScoreDisplay>Your best Score: {playerScore.bestScore}</PlayerScoreDisplay>
      <PlayerScoreDisplay>
        You played {playerScore.gamesPlay} games and won {playerScore.multiPlayersWin} times
      </PlayerScoreDisplay>
    </PlayerScoreWrapper>
  )) ||
  null;

export const mapStateToProps = (state: State) => ({
  playerScore: state.app.playerScore
});

export default connect(mapStateToProps)(PlayerScore);
