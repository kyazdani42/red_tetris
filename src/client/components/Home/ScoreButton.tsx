import * as React from 'react';
import { connect } from 'react-redux';

import { getScores } from '../../redux/actions';
import { ScoreButtonStyle } from './styles';

interface Props {
  dispatchGetScores: () => void;
}

export const ScoreButton: React.SFC<Props> = ({ dispatchGetScores }) => (
  <ScoreButtonStyle onClick={() => dispatchGetScores()} />
);

const mapDispatchToProps = (dispatch: any) => ({
  dispatchGetScores: () => dispatch(getScores())
});

export default connect(undefined, mapDispatchToProps)(ScoreButton);
