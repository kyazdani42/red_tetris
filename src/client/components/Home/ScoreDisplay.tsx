import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../redux/store';

import { setScores } from '../../redux/actions';
import { ModalWrapper, ScoreRowStyle, ScoreRowWrapper, ScoresWrapper, ScoreTitle } from './styles';

interface Props {
  scores: BestScore[] | null;
  resetScores: () => void;
}

const handleClick = (resetScores: () => void) => (e: any) => {
  if (e.target.classList.contains('modal')) {
    resetScores();
  }
};

export const ScoreDisplay: React.SFC<Props> = ({ scores, resetScores }) => scores && (
  <ModalWrapper className="modal" onClick={handleClick(resetScores)}>
    <ScoresWrapper>
      <ScoreTitle>Top 25 Scores</ScoreTitle>
      <ScoreRowWrapper>
        <ScoreRowStyle>
          <span><b>Player</b></span>
          <span><b>Top Score</b></span>
          <span><b>Games Played</b></span>
          <span><b>Multiplayer Game Wins</b></span>
        </ScoreRowStyle>
        {getRows(scores)}
      </ScoreRowWrapper>
    </ScoresWrapper>
  </ModalWrapper>
) || null;

const getRows = (scores: BestScore[]) => scores.map(
  d => (
  <ScoreRowStyle>
    <span>{d.name}</span>
    <span>{d.bestScore}</span>
    <span>{d.gamesPlay}</span>
    <span>{d.multiPlayersWin}</span>
  </ScoreRowStyle>
  )
);

const mapStateToProps = (state: State) => ({
  scores: state.app.scores
});

const mapDispatchToProps = (dispatch: any) => ({
  resetScores: () => dispatch(setScores(null))
});

export default connect(mapStateToProps, mapDispatchToProps)(ScoreDisplay);
