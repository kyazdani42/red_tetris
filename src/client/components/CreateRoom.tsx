import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { createRoom } from '../actions/actions';
import { CreateNameModal } from './CreateNameModal';

const Button = styled.div`
  font-family: ${props => props.theme.fonts.title};
  background-color: ${props => props.theme.colors.primary.dark};
  border-radius: 24px;
  color: #fff;
  line-height: 50px;
  height: 50px;
  width: 150px;
  font-size: 22px;
  border: 1px solid rgba(0, 0, 0, 0.5);
  text-align: center;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.2), 0 3px 6px rgba(0, 0, 0, 0.3);
  user-select: none;
  cursor: pointer;
  :hover {
    opacity: 0.94;
  }
`;

const Wrapper = styled.div`
  margin-top: 12px;
  display: flex;
  justify-content: center;
  align-items: flex-end;
`;

interface Props {
  dispatchCreateRoom: () => void;
}

export const CreateRoomButton: React.SFC<Props> = ({ dispatchCreateRoom }) => {
  const [displayModal, setDisplayModal] = React.useState(false);
  return (
    <React.Fragment>
    <Wrapper>
      <Button onClick={handleClick(setDisplayModal)}>
        New Game
      </Button>
    </Wrapper>
    {displayModal ? <CreateNameModal dispatchCreateRoom={dispatchCreateRoom} /> : null}
    </React.Fragment>
  );
};

const handleClick = (setDisplayModal: any) => (e: any) => {
  const target = e.currentTarget;
  target.style.opacity = '0.90';
  setTimeout(() => { target.style.opacity = '1'; }, 100);
  setDisplayModal(true);
};

const mapDispatchToProps = (dispatch: any) => ({
  dispatchCreateRoom: (username: string) => dispatch(createRoom(username))
});

export default connect(undefined, mapDispatchToProps)(CreateRoomButton);
