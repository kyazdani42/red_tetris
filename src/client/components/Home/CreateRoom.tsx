import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { createRoom } from '../../actions/actions';
import { CreateNameModal } from './CreateNameModal';

interface Props {
  dispatchCreateRoom: (username: string) => void;
}

const Button = styled.div`
  font-family: ${props => props.theme.fonts.title};
  background-color: ${props => props.theme.colors.primary.dark};
  border-radius: 4px;
  color: #fff;
  width: 35%;
  padding: 5px;
  font-size: 22px;
  border: 1px solid rgba(0, 0, 0, 0.5);
  text-align: center;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.2), 0 3px 6px rgba(0, 0, 0, 0.3);
  user-select: none;
  cursor: pointer;
  :hover {
    opacity: 0.94;
  }
  @media screen and (max-width: 500px) {
    font-size: 18px;
  }
`;

const Wrapper = styled.div`
  margin-top: 12px;
  display: flex;
  justify-content: center;
  align-items: flex-end;
`;

export const CreateRoomButton: React.SFC<Props> = ({ dispatchCreateRoom }) => {
  const [displayModal, setDisplayModal] = React.useState(false);
  let modal;
  if (displayModal) {
    modal = <CreateNameModal setDisplayModal={setDisplayModal} dispatch={dispatchCreateRoom} />;
  } else {
    modal = null;
  }
  return (
    <React.Fragment>
    <Wrapper>
      <Button onClick={handleClick(setDisplayModal)}>
        New Game
      </Button>
    </Wrapper>
    {modal}
    </React.Fragment>
  );
};

const handleClick = (setDisplayModal: any) => (e: any) => {
  const target = e.currentTarget;
  target.style.opacity = '0.90';
  setTimeout(() => { target.style.opacity = '1'; }, 100);
  setDisplayModal(true);
};

const mapDispatchToProps = (dispatch: any) => ({
  dispatchCreateRoom: (username: string) => dispatch(createRoom(username))
});

export default connect(undefined, mapDispatchToProps)(CreateRoomButton);
