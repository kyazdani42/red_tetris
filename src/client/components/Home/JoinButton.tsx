import * as React from 'react';
import { connect } from 'react-redux';
import { withTheme } from 'styled-components';

import { joinRoom, setModal } from '../../redux/actions';
import { State } from '../../redux/store';
import CreateNameModal from './CreateNameModal';
import { JoinButtonStyle } from './styles';
import { handleClick } from './utils';

interface Props {
  dispatchJoinRoom: (id: string) => (name: string) => void;
  dispatchSetModal: (display: boolean) => void;
  displayModal: boolean;
  roomId: string;
}
export const JoinButton: React.SFC<Props> = ({
  dispatchJoinRoom,
  roomId,
  dispatchSetModal,
  displayModal
}) => {
  let modal;
  if (displayModal) {
    modal = (
      <CreateNameModal
        className="name-modal"
        setDisplayModal={dispatchSetModal}
        handleDispatch={dispatchJoinRoom(roomId)}
      />
    );
  } else {
    modal = null;
  }
  return (
    <React.Fragment>
      <JoinButtonStyle onClick={handleClick(dispatchSetModal)}>
        <span>Join</span>
        <Svg />
      </JoinButtonStyle>
      {modal}
    </React.Fragment>
  );
};

const Svg = () => (
  <svg width="16" height="16" viewBox="0 0 24 24">
    <path d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z" />
    <path fill="none" d="M0 0h24v24H0z" />
  </svg>
);

const mapStateToProps = (state: State, { roomId }: { roomId: string }) => ({
  roomId,
  displayModal: state.app.modal
});

const mapDispatchToProps = (dispatch: any) => ({
  dispatchJoinRoom: (id: string) => (name: string) =>
    dispatch(joinRoom({ room: id, playerName: name })),
  dispatchSetModal: (display: boolean) => dispatch(setModal(display))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTheme(JoinButton as any));
