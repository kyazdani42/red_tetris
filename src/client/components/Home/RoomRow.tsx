import * as React from 'react';
import styled from 'styled-components';

import JoinButton from './JoinButton';

const StyledRow = styled.div`
  height: 50px;
  box-sizing: border-box;
  background-color: #000;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  color: #fff;
  padding: 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  text-align: center;
  border-bottom: 1px solid rgba(255,255,255,0.1);
`;

export const RoomRow: React.SFC<RoomType> = (props: any) => {
  const { running, name, players } = props;
  console.log(props);
  // const ownerName: string = `owner: ${owner.slice(0, 2)}`;
  const join = running ? null : <JoinButton roomId={name} />;

  return (
    <StyledRow>
      <h3 style={{ margin: 0 }}>
        <span>{'whatever'}</span>
      </h3>
      <div>
        <span>{players} player{players > 1 ? 's' : ''} </span>
      </div>
      {join}
    </StyledRow>
  );
};
