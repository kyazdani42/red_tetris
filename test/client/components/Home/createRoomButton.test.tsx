// tslint:disable
import * as React from 'react';
import { shallow } from 'enzyme';

import { CreateRoomButton } from '../../../../src/client/components/Home/CreateRoom';

describe('testing the component', () => {
  const props = {
    dispatchCreateRoom: () => null,
    dispatchSetModal: () => null,
    displayModal: true,
  };
  const wrapper = shallow(<CreateRoomButton {...props} />);
  it('smoke tests the component', () => {
    expect(wrapper.update()).toMatchSnapshot();
  });
  it('checks that join button exists', () => {
    expect(wrapper.find('.name-modal').exists()).toBeTruthy();
  })
  it('checks that join button doesn\'t exists', () => {
    wrapper.setProps({ displayModal: false });
    expect(wrapper.find('.name-modal').exists()).toBeFalsy();
  })
});
