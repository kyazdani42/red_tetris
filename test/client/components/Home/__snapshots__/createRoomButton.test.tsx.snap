// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dispatchCreateRoom={[Function]}
    dispatchSetModal={[Function]}
    displayModal={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED
          onClick={[Function]}
        >
          New Game
        </UNDEFINED>,
        <Connect(Component)
          className="name-modal"
          handleDispatch={[Function]}
          setDisplayModal={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "New Game",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "New Game",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": false,
            "rules": Array [
              "
  font-family: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  border-radius: 4px;
  color: #fff;
  width: 35%;
  padding: 4px;
  font-size: 22px;
  border: 1px solid rgba(0, 0, 0, 0.5);
  text-align: center;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.2), 0 3px 6px rgba(0, 0, 0, 0.3);
  user-select: none;
  cursor: pointer;
  :hover {
    opacity: 0.94;
  }
  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "name-modal",
          "handleDispatch": [Function],
          "setDisplayModal": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED
            onClick={[Function]}
          >
            New Game
          </UNDEFINED>,
          <Connect(Component)
            className="name-modal"
            handleDispatch={[Function]}
            setDisplayModal={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "New Game",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "New Game",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": false,
              "rules": Array [
                "
  font-family: ",
                [Function],
                ";
  background-color: ",
                [Function],
                ";
  border-radius: 4px;
  color: #fff;
  width: 35%;
  padding: 4px;
  font-size: 22px;
  border: 1px solid rgba(0, 0, 0, 0.5);
  text-align: center;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.2), 0 3px 6px rgba(0, 0, 0, 0.3);
  user-select: none;
  cursor: pointer;
  :hover {
    opacity: 0.94;
  }
  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "name-modal",
            "handleDispatch": [Function],
            "setDisplayModal": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
